xdg_shell.c

```diff
@@ -95,14 +95,14 @@ popup_unconstrain(struct cg_xdg_popup *popup)
 	struct cg_view *view = popup->view_child.view;
 	struct wlr_output *output = view->server->output->wlr_output;
 
-	int width, height;
-	wlr_output_effective_resolution(output, &width, &height);
+	struct wlr_box *output_box = wlr_output_layout_get_box(
+		view->server->output_layout, output);
```

Comment:
Can you extract `view->server->output` into its own variable (like is done for `output`), so that this fits on one line?